github-385-pr: Add Ansible Rulebook to glossary

michellemacrh opened the pull request:
PR for [issue 385](https://github.com/redhat-documentation/supplementary-style-guide/issues/385)

Add "Ansible Rulebook" to the glossary

bburt-rh commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/a.adoc:
```diff
@@ -343,6 +343,21 @@ Examples:
 
+[[ansible-rulebook]]
+==== image:images/yes.png[yes] Ansible Rulebook (noun)
+*Description*: An _Ansible Rulebook_ tells Event-Driven Ansible what sources to monitor for an event and what to do when certain conditions are met. Rulebooks are written in YAML, and are used like Ansible Playbooks. Write as shown: uppercase A and uppercase R. When using the term _rulebook_ without the Ansible prefix, use lowercase r.
```
```suggestion
*Description*: An _Ansible Rulebook_ tells Event-Driven Ansible what sources to monitor for an event and what to do when certain conditions are met. Rulebooks are written in YAML and are used like Ansible Playbooks. Write as shown: uppercase _A_ and uppercase _R_. When using the term _rulebook_ without the Ansible prefix, use lowercase _r_.
```
- Should `Event-Driven` be capitalized? I can't find it listed as an official product name in our product name list: https://docs.google.com/spreadsheets/d/1DLS_lS3VKidgZIvcLmLp9BoiqptkvqHWfe1D5FD2kfk/edit#gid=1259317633
- remove unnecessary comma following the subordinating conjunction `and`
- Follow guidelines for references to letters as letters per ISG: https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#letters-as-letters

michellemacrh replied:
Thanks for the feedback @bburt-rh !

- I found Event-Driven Ansible under the "Components, portals, and tools" tab (row 183) of the sheet you referred to but don't see a specific entry for "Event-Driven" on its own. Should we keep it capitalized when referring to "Event-Driven Ansible"? Happy to change if needed.
- Good catch TY on the comma and the letters as letters guideline.

bburt-rh replied:
@michellemacrh - Good catch! Didn't see it on the other tab. I was looking only on the main product name tab of the spreadsheet.

michellemacrh commented:
I've added the following as glossary entries based on feedback in the forum-ccs-style Slack channel: 
- rulebook
- playbook

I also added links between each entry, for example under "Ansible Rulebook" there's a link to "rulebook" and vice versa.

bburt-rh commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:
```diff
@@ -228,6 +228,22 @@ When combined with privileges, permissions define a user's or resource's overall
 
+[[playbook]]
+==== image:images/yes.png[yes] playbook (noun)
+*Description*: Playbooks are Ansible's configuration, deployment, and orchestration language.
```
```suggestion
*Description*: Playbooks are the configuration, deployment, and orchestration language for Ansible Automation Platform.
```
- Per ISG, "Do not use 's to show the possessive form of brand names or product names" - https://www.ibm.com/docs/en/ibm-style?topic=references-product-service-names#general-guidelines"

bburt-rh commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:
```diff
@@ -228,6 +228,22 @@ When combined with privileges, permissions define a user's or resource's overall
+[[playbook]]
+==== image:images/yes.png[yes] playbook (noun)
+*Description*: Playbooks are Ansible's configuration, deployment, and orchestration language.
+Playbooks can describe a policy you want your remote systems to enforce, or a set of steps in a general IT process.
```
```suggestion
Playbooks can describe a policy you want your remote systems to enforce or a set of steps in a general IT process.
```
- Remove comma after subordinating conjunction `or`. A comma would only be necessary if  an independent clause followed `or`.

michellemacrh replied:
Another copy over from "Ansible Playbook". I'll update that entry also.

bburt-rh commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:
```diff
@@ -228,6 +228,22 @@ When combined with privileges, permissions define a user's or resource's overall
+
+Examples:
+
+* Run a playbook in Ansible.
```
```suggestion
* Run a playbook in Ansible.
* Run an Ansible Playbook.
```
- Consider adding an additional example so that you show both acceptable usages.

bergerhoffer commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:
```diff
@@ -228,6 +228,23 @@ When combined with privileges, permissions define a user's or resource's overall
 
+[[playbook]]
+==== image:images/yes.png[yes] playbook (noun)
+*Description*: Playbooks are the configuration, deployment, and orchestration language for Ansible Automation Platform.
```
```suggestion
*Description*: _Playbooks_ are the configuration, deployment, and orchestration language for Ansible Automation Platform.
```

bergerhoffer reviewed (commented):
Just a few suggestions. Thanks again for expanding to add playbook and rulebook!

michellemacrh commented:
hhmm.. I seem to have pulled in some additional changes with that most recent commit. Is this correct? 

The commit in my forked repo looks to be correct: [Forked Repo Commit](https://github.com/michellemacrh/supplementary-style-guide/commit/d1a7e587137fa33f813114d4730b226d726e3886)

michellemacrh commented:
Feedback implemented per @bergerhoffer. 
Had some issues with updating the PR so needed to do an interactive rebase to remove additional commits that were outside of this PR.

michellemacrh commented:
No worries at all @bergerhoffer! Thank you for your patience while I navigated some fun Git issues 😅

bergerhoffer reviewed (approved):
LGTM!

bredamc commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/r.adoc:
```diff
@@ -1079,6 +1079,21 @@ Use the name of the tab when you refer to it, for example, "the *Storage* tab".
+
+Examples:
+
+* Use a rulebook in Ansible.
```
The "playbook" entry has the same two examples as the "Ansible Playbook" entry.

However, the "rulebook" entry has only one example. Should we add the other example from the "Ansible Rulebook" entry?

bburt-rh replied:
I would say it's fine to leave the one example because the `Ansible Rulebook` usage is not mentioned in the Description. In the other entry, both usages are mentioned in the description. I suppose you could expand the description here so that it's parallel with the other entry, and then add the extra example. But if you leave it as-is, then imo no extra example is needed.

bburt-rh replied:
It wouldn't hurt to add it, per @bredamc's suggestion, because the two usages are closely related.

michellemacrh commented:
Thanks for the feedback :) I've added another example under the rulebook entry.

bredamc commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/r.adoc:
```diff
@@ -1079,6 +1079,22 @@ Use the name of the tab when you refer to it, for example, "the *Storage* tab".
+Examples:
+
+* Use a rulebook in Ansible.
+* Use an Ansible Playbook.
```
```suggestion
* Use an Ansible Rulebook.
```

I think you meant Rulebook, not Playbook? :)

michellemacrh replied:
Oh gosh, so sorry! Forgot to wake up this morning 😅  I'll correct it!

michellemacrh replied:
OK that should be updated! Apologies again and thanks for catching that @bredamc :)

bredamc reviewed (approved):
LGTM :)

mportman12 commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/a.adoc:
```diff
@@ -342,7 +342,22 @@ Examples:
+
+[[ansible-rulebook]]
+==== image:images/yes.png[yes] Ansible Rulebook (noun)
+*Description*: An _Ansible Rulebook_ tells Event-Driven Ansible what sources to monitor for an event and what to do when certain conditions are met. Rulebooks are written in YAML and are used like Ansible Playbooks. Write as shown: uppercase _A_ and uppercase _R_. When using the term _rulebook_ without the Ansible prefix, use lowercase _r_.
```
```suggestion
*Description*: An _Ansible Rulebook_ tells Event-Driven Ansible which sources to monitor for an event and what to do when certain conditions are met. Rulebooks are written in YAML and are used like Ansible Playbooks. Write as shown: uppercase _A_ and uppercase _R_. When using the term "rulebook" without the Ansible prefix, use lowercase _r_.
```

mportman12 commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/r.adoc:
```diff
@@ -1079,6 +1079,22 @@ Use the name of the tab when you refer to it, for example, "the *Storage* tab".
+[[rulebook]]
+==== image:images/yes.png[yes] rulebook (noun)
+*Description*: A _rulebook_ is a YAML file containing a list of rules, event source definitions, and hosts to run against. 
+Rulebooks are used in Event-Driven Ansible. When using the term _rulebook_ without the Ansible prefix, use lowercase _r_.
```
```suggestion
Rulebooks are used in Event-Driven Ansible. When using the term "rulebook" without the Ansible prefix, use lowercase _r_.
```

mportman12 reviewed (changes requested):
Left a few small suggestions.

michellemacrh commented:
Updated based on feedback.

mportman12 commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:
```diff
@@ -228,6 +228,23 @@ When combined with privileges, permissions define a user's or resource's overall
+==== image:images/yes.png[yes] playbook (noun)
+*Description*: _Playbooks_ are the configuration, deployment, and orchestration language for Ansible Automation Platform.
+Playbooks can describe a policy you want your remote systems to enforce or a set of steps in a general IT process.
+When using the term _playbook_ without the Ansible prefix, use lowercase _p_.
```
```suggestion
When using the term "playbook" without the Ansible prefix, use lowercase _p_.
```

mportman12 reviewed (changes requested):
One tiny change

bergerhoffer reviewed (approved):
LGTM

mportman12 reviewed (approved):
LGTM

bburt-rh commented:
LGTM. Merging.
